src/libraries/System.Security.Cryptography/tests/OpenSslNamedKeysTests.manual.cs

```diff
@@ -174,17 +225,202 @@ public static void Engine_UsePublicKey()
+            RSAEncryptionPadding padding = RSAEncryptionPadding.Pkcs1;
+            using SafeEvpPKeyHandle priKeyHandle = SafeEvpPKeyHandle.OpenKeyFromProvider(Tpm2ProviderName, TpmRsaDecryptKeyHandleUri);
+            using RSA rsaPri = new RSAOpenSsl(priKeyHandle);
+            byte[] rsaPubBytes = rsaPri.ExportSubjectPublicKeyInfo();
+            RSA rsaPub = RSA.Create();
+            rsaPub.ImportSubjectPublicKeyInfo(rsaPubBytes, out int bytesRead);
+            Assert.Equal(rsaPubBytes.Length, bytesRead);
+
+            using RSA rsaBad = RSA.Create();
+            rsaBad.KeySize = rsaPri.KeySize;
+
+            byte[] data = new byte[] { 1, 2, 3, 1, 1, 2, 3 };
+            byte[] encryptedWithDifferentKey = rsaBad.Encrypt(data, padding);
+            Assert.ThrowsAny<CryptographicException>(() => rsaPri.Decrypt(encryptedWithDifferentKey, padding));
+
+            // TPM private key is intended only for decrypt
+            Assert.ThrowsAny<CryptographicException>(() => rsaPri.Encrypt(data, padding));
+
+            // can use same key more than once
+            for (int i = 0; i < 10; i++)
+            {
+                data[0] = (byte)i;
+                byte[] encrypted = rsaPub.Encrypt(data, padding);
+                Assert.NotEqual(encrypted, data);
+
+                Assert.Equal(data, rsaPri.Decrypt(encrypted, padding));
+            }
+        }
+
+        // Sanity tests for flags
+        [ConditionalFact(nameof(ShouldRunEngineTests))]
+        public static void SanityTest_EngineAny()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunTpmTssTests))]
+        public static void SanityTest_EngineTpmTss()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunAnyProviderTests))]
+        public static void SanityTest_AnyTpm2Provider()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunProviderEcDsaTests))]
+        public static void SanityTest_Tpm2ProviderEcDsa()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunProviderEcDhTests))]
+        public static void SanityTest_Tpm2ProviderEcDh()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunProviderRsaSignTests))]
+        public static void SanityTest_Tpm2ProviderRsaSign()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
+
+        [ConditionalFact(nameof(ShouldRunProviderRsaDecryptTests))]
+        public static void SanityTest_Tpm2ProviderRsaDecrypt()
+        {
+            Assert.False(ShouldFailTests, "This test is supposed to fail");
+        }
```

Comment:
I don't think these should be added... they look like TODOs to me.

As Yoda might say, DO or DO-LETE, there is no "this test is supposed to fail".